autoload/ale/handlers/redpen.vim

```diff
@@ -0,0 +1,29 @@
+" Author: rhysd https://rhysd.github.io
+" Description: Redpen, a proofreading tool (http://redpen.cc)
+
+function! ale#handlers#redpen#HandleRedpenOutput(buffer, lines) abort
+    " Only one file was passed to redpen. So response array has only one
+    " element.
+    let l:res = json_decode(join(a:lines))[0]
+    let l:output = []
+    for l:err in l:res.errors
+        let l:item = {
+        \   'text': l:err.message . ' (' . l:err.validator . ')',
+        \   'type': 'W',
+        \}
+        if has_key(l:err, 'startPosition')
+            let l:item.lnum = l:err.startPosition.lineNum
+            let l:item.col = l:err.startPosition.offset + 1
+            if has_key(l:err, 'endPosition')
+                let l:item.end_lnum = l:err.endPosition.lineNum
+                let l:item.end_col = l:err.endPosition.offset
```

Comment:
Actually, won't you need to add `+ 1` here too? I imagine if the `startPosition` is off by 1, the `endPosition` will be too.

Reply:
No, we don't need. The range of error is `[start, end)`. So left edge is included but right edge is excluded. And it was revealed that the `offset` is 0-based. So we need to add 1 only to left edge.

Reply:
The `end_col` value has to be the column number the error ends on, so the range for `col` and `end_col` is `[start, end]`.

Reply:
The end offset does seem to be 1-based given the output, so maybe this is okay. For "plugin" it says the word is a spelling mistake from columns 10 to 15, so that checks out.

Reply:
oh, I see. I did not know that. Thank you.